core-common/src/main/java/org/zalando/nakadi/annotations/validation/DataLakeAnnotationValidator.java

```diff
@@ -30,6 +30,17 @@ public boolean isValid(final Map<String, String> annotations, final ConstraintVa
                         .addConstraintViolation();
                 return false;
             }
+            if (annotations.get(MATERIALISE_EVENTS_ANNOTATION).equals("on") &&
```

Comment:
Minor: `annotations.get(MATERIALISE_EVENTS_ANNOTATION)` could be stored in a local variable and reused in this method (instead of multiple `Map` method calls).